Jenkinsfile

```diff
@@ -1,68 +1,140 @@
-def NODE_NAME = "sw"
-def BRANCH_NAME_RELEASE = "release"
-def BRANCH_NAME_MASTER = "master"
+def SW_NODE = "windows-slave"
+def ECAT_NODE = "ecat-test"
+def CAN_NODE = "canopen-test"
 def PYTHON_VERSIONS = ["3.6", "3.7", "3.8", "3.9"]
 def style_check = false
 
-node(NODE_NAME)
+node(ECAT_NODE)
 {
     deleteDir()
-    if (env.BRANCH_NAME == BRANCH_NAME_MASTER || env.BRANCH_NAME.contains(BRANCH_NAME_RELEASE))
+
+    stage('Checkout') {
+        checkout scm
+    }
+
+    stage('Update FW to drives') {
+        bat """
+            python -m venv ingeniamotion
+            ingeniamotion\\Scripts\\python.exe -m pip install ingeniamotion ping3
+            ingeniamotion\\Scripts\\python.exe tests\\load_FWs.py ethercat
+        """
+    }
+
+    stage('Install deps') {
+        bat '''
+            python -m venv venv
+            venv\\Scripts\\python.exe -m pip install -r requirements\\test-requirements.txt
+        '''
+    }
+
+    stage('Run EtherCAT embedded tests') {
+        bat '''
+            venv\\Scripts\\python.exe -m pytest tests --protocol ethercat
+            exit /b 0
+        '''
+    }
+}
+
+node(CAN_NODE)
+{
+    deleteDir()
+
+    stage('Checkout') {
+        checkout scm
+    }
+
+    stage('Update FW to drives') {
+        bat """
+            python -m venv ingeniamotion
+            ingeniamotion\\Scripts\\python.exe -m pip install ingeniamotion ping3
+            ingeniamotion\\Scripts\\python.exe tests\\load_FWs.py canopen
+        """
+    }
+
+    stage('Install deps') {
+        bat '''
+            python -m venv venv
+            venv\\Scripts\\python.exe -m pip install -r requirements\\test-requirements.txt
+        '''
+    }
+
+    stage('Run CANopen tests') {
+        bat '''
+            venv\\Scripts\\python.exe -m pytest tests --protocol canopen
+            exit /b 0
+        '''
+    }
+
+    stage('Run Ethernet tests') {
+        bat '''
+            venv\\Scripts\\python.exe -m pytest tests --protocol ethernet
+            exit /b 0
+        '''
+    }
```

Comment:
We need to run no-connection tests as well:

```
    stage('Run no-connection tests') {
        bat '''
            venv\\Scripts\\python.exe -m pytest tests
            exit /b 0
        '''
    }
````

Reply:
Ok, I'll add it.